src/streaming/controllers/StreamController.js

```diff
@@ -138,15 +139,21 @@ function StreamController() {
      * Called when current playback position is changed.
      * Used to determine the time current stream is finished and we should switch to the next stream.
      */
-    function onPlaybackTimeUpdated(/*e*/) {
-
+    function onPlaybackTimeUpdated(e) {
         if (isVideoTrackPresent()) {
             const playbackQuality = videoModel.getPlaybackQuality();
             if (playbackQuality) {
                 metricsModel.addDroppedFrames(Constants.VIDEO, playbackQuality);
             }
         }
 
+        //detect end of stream for a multiperiod stream. Like timeUpdate could occurs at different time from the end, trigger PLAYBACK_ENDED when timeToEnd is less or equal to 0.5 second.
+        if (endedEventNeeded && e.timeToEnd <= 0.5) {
+            //send PLAYBACK_ENDED in order switch to a new period
+            eventBus.trigger(Events.PLAYBACK_ENDED);
```

Comment:
@nicosang, what do you think if we replace this line with this other one?
```
setTimeout(function () {
    eventBus.trigger(Events.PLAYBACK_ENDED);
}, e.timeToEnd > 0  ? e.timeToEnd : 0)
```

Another question. Could we ensure that in all the cases onPlaybackTimeUpdated is going to be called at least one time after onStreamBufferingCompleted?

Reply:
@epiclabsDASH , do you prefer the last commit with the use of a timer in onStreamBufferingCompleted function rather than onPlaybackTimeUpdated?

Reply:
I think your last change makes more sense, using the timeout in onStreamBufferingCompleted and avoiding the possibility of having race conditions between onPlaybackTimeUpdated and onStreamBufferingCompleted.

Let's wait for @spiterikevin opinion.

Thanks!